profiler/upload_test.go

```diff
@@ -48,57 +49,83 @@ func TestTryUpload(t *testing.T) {
 	defer func(cid string) { containerID = cid }(containerID)
 	containerID = ""
 
-	srv, srvURL, waiter := makeTestServer(t, 200)
-	defer srv.Close()
-	p, err := unstartedProfiler(
-		WithAgentAddr(srvURL.Host),
-		WithService("my-service"),
-		WithEnv("my-env"),
-		WithTags("tag1:1", "tag2:2"),
-	)
-	require.NoError(t, err)
-	err = p.doRequest(testBatch)
-	require.NoError(t, err)
-	header, fields, tags := waiter()
-
-	assert := assert.New(t)
-	assert.Empty(header.Get("Datadog-Container-ID"))
-	assert.ElementsMatch([]string{
-		"host:my-host",
-		"runtime:go",
-		"service:my-service",
-		"env:my-env",
-		"tag1:1",
-		"tag2:2",
-		fmt.Sprintf("pid:%d", os.Getpid()),
-		fmt.Sprintf("profiler_version:%s", version.Tag),
-		fmt.Sprintf("runtime_version:%s", strings.TrimPrefix(runtime.Version(), "go")),
-		fmt.Sprintf("runtime_compiler:%s", runtime.Compiler),
-		fmt.Sprintf("runtime_arch:%s", runtime.GOARCH),
-		fmt.Sprintf("runtime_os:%s", runtime.GOOS),
-		fmt.Sprintf("runtime-id:%s", globalconfig.RuntimeID()),
-	}, tags)
-	for k, v := range map[string]string{
-		"format":   "pprof",
-		"runtime":  "go",
-		"types[0]": "cpu",
-		"data[0]":  "my-cpu-profile",
-		"types[1]": "alloc_objects,alloc_space",
-		"data[1]":  "my-heap-profile",
-	} {
-		assert.Equal(v, fields[k], k)
+	fixtures := []struct {
+		description     string
+		startTestServer startTestServerFn
+		exopts          func(string) []Option
```

Comment:
This whole logic and abstraction is way overkill and really hard to wrap one's head around. It's just two tests. I don't think we need _this much_ abstraction for them.

Can we leave the old HTTP server as it were and simply add an option or create a separate one for UDS?

Reply:
The test itself reads easily enough IMO, but I'm happy to dup a bunch of code to get past review.
Because the types don't align, have to duplicate if we aren't willing to use simple callbacks.
Adding flags like `uds bool` is bad smell.  We'll just duplicate it.